flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/hybrid/HsFullSpillingStrategyTest.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.partition.hybrid;
+
+import org.apache.flink.runtime.io.network.partition.hybrid.HsSpillingStrategy.Action;
+import org.apache.flink.runtime.io.network.partition.hybrid.HsSpillingStrategy.NeedLockAction;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.IntStream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+/** Tests for {@link HsFullSpillingStrategy}. */
+class HsFullSpillingStrategyTest extends HsSpillingStrategyTestBase {
+    public static final int NUM_BUFFERS_TRIGGER_SPILLING = 10;
+
+    @BeforeEach
+    void before() {
+        this.spillStrategy =
+                new HsFullSpillingStrategy(
+                        HybridShuffleConfiguration.builder(NUM_SUBPARTITIONS, 1)
+                                .setFullStrategyNumBuffersTriggerSpilling(
+                                        NUM_BUFFERS_TRIGGER_SPILLING)
+                                .build());
+    }
+
+    @Override
+    @Test
+    void testOnBufferFinished() {
+        testUnSpillBufferBelowThreshold();
+        testUnSpilledBufferEqualThreshold();
```

Comment:
It is not recommended to combine multiple test cases.

Reply:
Splits these into single methods.